server.js

```diff
@@ -0,0 +1,140 @@
+var fs = require('fs');
+var path = require('path');
+var bodyParser = require('body-parser');
+var jsonParser = bodyParser.json();
+var express = require('express');
+var app = express();
+var datastore = './data';
+
+function renderJSON(res, data) {
+    res.status(200)
+        .set('Content-Type', 'application/json')
+        .json(data);
+}
+
+function filterJSONFiles(data) {
+    return data.filter(function(file) {
+        return (file.substr(-5) === '.json') ? 1 : 0;
+    });
+}
+
+app.route('/elements')
+
+    // List all elements.
+    .get(function(req, res, next) {
+        console.log('LIST ALL ELEMENTS - GET /elements');
+        var output = [];
+
+        // Get a list of all files in the datastore. Filter the 'files' for those that are '.json'. Read in the contents
+        // of all JSON files and store their content into the 'output' array.
+        fs.readdir(datastore, function(err, files) {
+            if (err) { next(err); }
+            files = filterJSONFiles(files);
+            files.forEach(function(file) {
+                fs.readFile(path.join(datastore, '/', file), { encoding: 'utf8' }, function(err, data) {
+                    if (err) { next(err); }
+                    output.push(JSON.parse(data));
+                    if (output.length === files.length) { renderJSON(res, output); }                       
+                });
+            });
+        });
+    })
+
+    // Create a piece of data.
+    .post(jsonParser, function(req, res, next) {
+        console.log('POST to /elements - CREATING DATA');
+        var element = req.body.element || '';
+        var created = req.body.created || +new Date();
+        var dataToWrite = JSON.stringify({
+            element: element,
+            created: created
+        });
+
+        fs.readdir(datastore, function(err, files) {
+            if (err) { next(err); }
+            var fileNumber = files.filter(function(file) {
+                return (file.substr(-5) === '.json') ? 1 : 0;
+            });
+            fileNumber = fileNumber.length + 1;
+            fs.writeFile(datastore + '/' + fileNumber + '.json', dataToWrite, { encoding: 'utf8' }, function(err) {
+                if (err) { next(err); }
+                renderJSON({status: 'success', operation: 'POST', _id: fileNumber});
+            });    
+        });
+    });
+
+app.route('/elements/:id')
```

Comment:
You should modularize this (this should go into another file under a routes folder)